Domain/Problem/src/main/java/tw/waterball/judgegirl/problem/domain/repositories/ProblemQueryParams.java

```diff
@@ -31,17 +31,23 @@
 
     private final boolean excludeArchive;
 
+    private final boolean includeArchive;
+
     private final boolean includeInvisibleProblems;
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean includeInvisibleProblems) {
-        this(tags, page, true, includeInvisibleProblems);
+    private final boolean excludeVisibleProblems;
+
+    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean includeInvisibleProblems, boolean excludeVisibleProblems, boolean includeArchive) {
+        this(tags, page, true, includeInvisibleProblems, excludeVisibleProblems, includeArchive);
     }
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean excludeArchive, boolean includeInvisibleProblems) {
+    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean excludeArchive, boolean includeInvisibleProblems, boolean excludeVisibleProblems, boolean includeArchive) {
         this.tags = tags;
         this.page = page;
         this.excludeArchive = excludeArchive;
+        this.includeArchive = includeArchive;
         this.includeInvisibleProblems = includeInvisibleProblems;
+        this.excludeVisibleProblems = excludeVisibleProblems;
```

Comment:
This really challenges my logic. 😵‍💫 
I guess `boolean excludeArchive` and `boolean includeArchive` are always true/false or false/true, so we can combine them to `boolean archive` only.
Ditto for `includeInvisibleProblems` and `excludeVisibleProblems`.

Or you can share your idea about this.

Reply:
boolean archive
boolean visible
boolean invisible
```java
if (archive) {
    archive && (v || inV)
    // archive => archive: true , v: false, inV: false
} else {
    (v || inV)
    // problems => archive: false , v: true, inV: false
    // invisible => archive: false , v: false, inV: true
}
```